packages/flutter_tools/lib/src/build_runner/devfs_web.dart

```diff
@@ -339,8 +342,9 @@ class WebAssetServer implements AssetReader {
     }
 
     if (!file.existsSync()) {
-      final String webPath = globals.fs.path.join(
-        globals.fs.currentDirectory.childDirectory('web').path, requestPath);
+      final Uri webPath = globals.fs.currentDirectory
+        .childDirectory('web')
+        .uri.resolve(requestPath);
```

Comment:
This is weird, I've been able to download the "pubspec.yaml" or some "json" files that are in the root of the app (siblings of `web`).

(Try: `localhost:port/pubspec.yaml`, or some `.iml`)

I haven't been able to escape outside of the root of the app!

Reply:
Its actually from above where we look for dart source. We need to resolve down from the project directory to handle requests like lib/main.dart for source maps, so this is working as expected I think

Reply:
Ahhh, makes sense. Thanks for the clarification!